desc/magnetic_fields.py

```diff
@@ -1819,9 +1836,875 @@ def from_surface(
+                    given `source_grid`. An array of length `source_grid.num_nodes` if
+                    `scan=False`, or list of arrays, with list length `nscan`,
+                    if `scan=True`, corresponding to the list of `alpha`
+
+
+        """
+        assert (
+            int(current_helicity) == current_helicity
+        ), "current_helicity must be an integer!"
+        # maybe it is an EquilibriaFamily
+        if hasattr(eq, "__len__"):
+            eq = eq[-1]
+
+        # ensure vacuum eq, as we don't yet support finite beta
+        pres = np.max(np.abs(eq.compute("p")["p"]))
+        curr = np.max(np.abs(eq.compute("current")["current"]))
+        warnif(
+            pres > 1e-8,
+            UserWarning,
+            f"Pressure is non-zero (max {pres} Pa), "
+            + "finite beta not supported yet.",
+        )
+        warnif(
+            curr > 1e-8,
+            UserWarning,
+            f"Current is non-zero (max {curr} A), "
+            + "finite plasma currents not supported yet.",
+        )
+
+        if external_field:  # ensure given field is an instance of _MagneticField
+            assert isinstance(external_field, _MagneticField), (
+                "Expected"
+                + "MagneticField for argument external_field,"
+                + f" got type {type(external_field)} "
+            )
+        if source_grid is None:
+            source_grid = LinearGrid(M=30, N=30, NFP=self.NFP)
+        if eval_grid is None:
+            eval_grid = LinearGrid(M=30, N=30, NFP=self.NFP, sym=eq.sym)
+        if normalize:
+            B_eq_surf = eq.compute("|B|", eval_grid)["|B|"]
+            # just need it for normalization, so do a simple mean
+            normalization_B = jnp.mean(B_eq_surf)
+        else:
+            normalization_B = 1
+
+        # plasma surface normal vector magnitude on eval grid
+        ne_mag = eq.compute(["|e_theta x e_zeta|"], eval_grid)["|e_theta x e_zeta|"]
+        # winding surface normal vector magnitude on source grid
+        ns_mag = self.compute(["|e_theta x e_zeta|"], source_grid)["|e_theta x e_zeta|"]
+
+        # change self basis to match desired resolution
+        # TODO: allow cos for this as well?
+        if sym_Phi is None:
+            sym_Phi = "sin" if eq.sym else False
+        self.change_Phi_resolution(M=M_Phi, N=N_Phi, sym_Phi=sym_Phi)
+
+        # calculate net enclosed poloidal and toroidal currents
+        G_tot = -(eq.compute("G", grid=source_grid)["G"][0] / mu_0 * 2 * jnp.pi)
+
+        if external_field:
+            # calculate the portion of G provided by external field
+            # by integrating external toroidal field along a curve of constant theta
+            try:
+                G_ext = external_field.G
+            except AttributeError:
+                curve_grid = LinearGrid(
+                    N=eq.NFP * 1000,
+                    theta=jnp.array(jnp.pi),
```

Comment:
since we are not considering the plasma current at all, I dont think the specific curve matters, so long as it is taken at constant theta. (because we are not dotting the plasma field, just the field from the external coils)

I tested this for a few different theta and it seemed to work fine regardless of which one I chose.